src/components/Search.tsx

```diff
@@ -6,26 +8,79 @@ import * as SearchActions from '@libs/actions/Search';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import Log from '@libs/Log';
 import * as SearchUtils from '@libs/SearchUtils';
+import type {SearchColumnType, SortOrder} from '@libs/SearchUtils';
 import Navigation from '@navigation/Navigation';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
 import EmptySearchView from '@pages/Search/EmptySearchView';
 import useCustomBackHandler from '@pages/Search/useCustomBackHandler';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {SearchQuery} from '@src/types/onyx/SearchResults';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
 import SelectionList from './SelectionList';
 import SearchTableHeader from './SelectionList/SearchTableHeader';
+import type {TransactionListItemType} from './SelectionList/types';
 import TableListItemSkeleton from './Skeletons/TableListItemSkeleton';
 
+const columnNamesToPropertyMap = {
+    [CONST.SEARCH_TABLE_COLUMNS.TO]: 'formattedTo' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.FROM]: 'formattedFrom' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.DATE]: 'date' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TAG]: 'tag' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.MERCHANT]: 'merchant' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TOTAL]: 'formattedTotal' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.CATEGORY]: 'category' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TYPE]: 'type' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.ACTION]: 'action' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.DESCRIPTION]: null,
+    [CONST.SEARCH_TABLE_COLUMNS.TAX_AMOUNT]: null,
+};
+
+function getSortedData(data: TransactionListItemType[], sortBy?: SearchColumnType, sortOrder?: SortOrder) {
```

Comment:
I think we should move this to a utils file so we don't clutter the component with a ton of business logic.